zeroforcing/metagraph.pyx

```diff
@@ -0,0 +1,252 @@
+import itertools
+
+from sage.data_structures.bitset cimport (
+    Bitset,
+    FrozenBitset
+)
+
+from sage.data_structures.bitset_base cimport (
+    bitset_add,
+    bitset_clear,
+    bitset_copy,
+    bitset_difference,
+    bitset_free,
+    bitset_in,
+    bitset_init,
+    bitset_intersection,
+    bitset_isempty,
+    bitset_len,
+    bitset_next,
+    bitset_remove,
+    bitset_t,
+    bitset_union
+)
+
+from cysignals.memory cimport (
+    sig_free,
+    sig_malloc
+)
+
+from fastqueue cimport FastQueueForBFS
+
+
+cdef class ZFSearchMetagraph:
+    def __cinit__(self, graph_for_zero_forcing):
+        self.num_vertices = graph_for_zero_forcing.num_verts()
+        self.neighborhood_array = <bitset_t*> sig_malloc(self.num_vertices*sizeof(bitset_t)) #ALLOCATE NEIGHBORHOOD_ARRAY
+        
+        
+        # Initialize/clear extend_closure bitsets
+        bitset_init(self.filled_set, self.num_vertices)
```

Comment:
Maybe have a class property called `self.bitsets` or something that returned a list of all bitsets so that code like this (and similarly in other spots) can be reduced to:

```python
for bitset in self.bitsets:
    bitset_init(bitset, self.num_vertices)
```